assets/js/search.js

```diff
@@ -0,0 +1,101 @@
+---
+---
```

Comment:
Might want to document that the `---` lines are necessary to ensure that `{{ site.baseurl }}` is substituted by Jekyll.

And, in general, mind giving a brief overview of the flow of the callbacks here? Also, each of the functions could use a brief descriptive comment.